### PR TITLE
feat(cosh): support multi-hook toggles

### LINKED ISSUE #1200: [cosh] bug: cosh在disable hooks的时候，不可以支持同时选择多个
### Component

cosh

### What happened?

cosh在disable hooks的时候，不可以同时选择多个

在实际使用中，不想用安全的组件，需要disable多次。比较麻烦，希望能同时选择多个去disable。提高使用体验

Image

### Diagnostic Output

```shell

```

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by kongche-jbw**
## Description

This PR adds multi-hook enable and disable support for `/hooks enable` and `/hooks disable`. Explicit arguments now support space-separated, comma-separated, and mixed hook names with deduplication, while registry-aware parsing preserves existing hooks whose fallback names contain spaces. When no hook names are provided, the interactive UI opens a multi-select prompt for the relevant enabled or disabled hooks.

## Related Issue

closes #1200

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional change)
- [ ] Performance improvement
- [ ] CI/CD or build changes

## Scope

- [x] `cosh` (copilot-shell)

## Checklist

- [x] My code follows the project's code style
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] For `cosh`: Lint passes, type check passes, and tests pass
- [x] Lock files are up to date (`package-lock.json` / `Cargo.lock`)

## Testing

- `npm run test --workspace=packages/cli -- hooksCommand HookMultiSelectPrompt`
  - Passed: 26 tests

- `npm run typecheck --workspace=packages/cli`
  - Failed on existing unrelated errors around `logSessionSummary` exports and `SessionMetrics.toolErrorCounts`.
  - The failures do not point to files changed by this PR.

## Additional Notes

The second-and-later argument completion limitation remains out of scope because it belongs to the global slash completion parser rather than the `/hooks` command implementation.

### CODE REVIEW

File: src/copilot-shell/packages/cli/src/ui/commands/hooksCommand.ts [hunk 1]
```diff
@@ -12,7 +12,76 @@ import type {
...
+    if (knownNames.has(part)) {
+      result.push(part);
+    } else {
+      for (const token of part.split(/\s+/)) {
+        if (token) result.push(token);
+      }
+    }
+  }
+
+  return [...new Set(result)];
```
Comment (samchu-zsl) [inline src/copilot-shell/packages/cli/src/ui/commands/hooksCommand.ts:51]: 🟡 **Unknown hook names leak through silently.**

`resolveHookNames` happily returns any token that does not match a known hook (the inner `for (const token of part.split(/\s+/))` only deduplicates, it does not filter against `knownNames`). Combined with `HookRegistry.setHookEnabled` — which only logs a `debugLogger.warn` for unmatched names — this leads to a false-positive UX:

```
/hooks disable lin test     # `lin` is a typo
→ "Disabled 2 hooks for this session: lin, test"
```

The success message lists names that did nothing. Consider one of:
- filter `result` against `knownNames` after tokenization, and surface unknown names to the caller (e.g. `"Disabled 1 hook: test (unknown: lin)"`), or
- have `toggleHooks` return both the requested and the actually-affected name sets and adjust the message accordingly.

The single-name path had the same blind spot pre-PR, but multi-name amplifies it because a single typo no longer surfaces as the only requested name.

File: src/copilot-shell/packages/cli/src/ui/commands/hooksCommand.ts [hunk 2]
```diff
@@ -12,7 +12,76 @@ import type {
...
+
+  for (const name of names) {
+    registry.setHookEnabled(name, enabled);
+  }
+
+  if (names.length === 1) {
+    return {
+      type: 'message',
+      messageType: 'info',
+      content: t('Hook "{{name}}" has been {{action}} for this session.', {
```
Comment (samchu-zsl) [inline src/copilot-shell/packages/cli/src/ui/commands/hooksCommand.ts]: 🟡 **New i18n keys missing from locale files.**

This line, plus the multi-name template on L78 and the prompt strings in `HookMultiSelectPrompt.tsx`, introduce keys that exist nowhere in `packages/cli/src/i18n/locales/en.js` or `zh.js`:

```
'Hook "{{name}}" has been {{action}} for this session.'
'{{verb}} {{count}} hooks for this session: {{names}}'
'No disabled hooks to enable.'
'No enabled hooks to disable.'
'Select hooks to enable'
'Select hooks to disable'
'Enable disabled hooks'
'Disable active hooks'
'{{count}} selected'
'↑↓/jk navigate, Space toggle, a select all, Enter confirm, Esc cancel'
```

Because `t()` falls back to the key, EN appears to work by accident, but ZH users see raw English. The previous keys `'Enable a disabled hook'` and `'Disable an active hook'` also become orphans in both locale files. Please add the new keys to both locales (and remove the orphans) so `npm run check-i18n` passes.

File: src/copilot-shell/packages/cli/src/ui/components/HookMultiSelectPrompt.tsx [hunk 3]
```diff
@@ -0,0 +1,194 @@
+/**
+ * @license
+ * Copyright 2025 Qwen
```
Comment (samchu-zsl) [inline src/copilot-shell/packages/cli/src/ui/components/HookMultiSelectPrompt.tsx]: 🟡 **Copyright header should be `Alibaba Cloud` for new files.**

Per the project's code standards, new files authored in 2026 use:

```
/**
 * @license
 * Copyright 2026 Alibaba Cloud
 * SPDX-License-Identifier: Apache-2.0
 */
```

Same applies to `HookMultiSelectPrompt.test.tsx` and `hooksCommand.test.ts` introduced in this PR (all currently say `Copyright 2025 Qwen`).

**Review by samchu-zsl — COMMENTED** [review]
## Summary

Nice expansion of `/hooks enable|disable` to multi-name plus a multi-select UI. Code is well-structured, exhaustive `never` switches are kept consistent, and tests cover the main argument-parsing edge cases. A few items worth addressing before merge.

### 🟡 Major
- **i18n parity is broken.** Ten new `t()` keys are introduced but neither `packages/cli/src/i18n/locales/en.js` nor `zh.js` is updated. Because `t()` falls back to the key, EN looks correct by accident, but ZH users will see English strings, the two retired keys (`'Enable a disabled hook'` / `'Disable an active hook'`) become orphans, and `npm run check-i18n` will flag this (it is the project's standard gate per cross-cutting #5 in `cosh-concerns`).
- **Unknown hook names are silently treated as success.** Any token returned by `resolveHookNames` that does not match a registered hook still flows into `registry.setHookEnabled`, which only emits `debugLogger.warn`. The user receives a green `"Disabled N hooks: ..."` even when none were affected. The previous single-name path had the same blind spot, but the multi-name version amplifies it. Suggest filtering / surfacing the unknown subset in the result message.
- **Copyright headers in new files use `Qwen`.** Project convention is `Copyright 2026 Alibaba Cloud` for new files authored in 2026. Affects `HookMultiSelectPrompt.tsx`, `HookMultiSelectPrompt.test.tsx`, and `hooksCommand.test.ts`.

### ℹ️ Info
- No entry added to `src/copilot-shell/CHANGELOG.md`. Engineering discipline asks for a user-perceivable change to ship with a one-line changelog bullet.
- The PR description acknowledges typecheck fails on "unrelated errors" and leaves the corresponding checklist box unchecked. Worth double-checking the failures are indeed unrelated, given the new union member `MultiSelectHooksActionReturn` ripples through every consumer of `SlashCommandActionReturn`.

### Positives
- Both exhaustive switches (`nonInteractiveCliCommands.ts` and `slashCommandProcessor.ts`) are updated, preserving `never` narrowing.
- ACP path is implicitly covered via the non-interactive `unsupported` mapping.
- Tests exercise dedup, mixed separators, the "hook name containing spaces" preservation, and the multi-select callback path.

**Review by samchu-zsl — APPROVED** [review]
LGTM, thx!
